File: server-mock/src/main/java/com/belogrudov/servermock/controller/Subject.java
package com.belogrudov.servermock.controller;

import java.time.LocalDate;
import java.util.List;

import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class Subject {
    String uuid;
    String name;
    String surname;
    LocalDate birthdate;
    String countryCode;
    String bio;
    List<Promise> promises;

    public String toSimpleJson() {
        return """
                {
                    "uuid":"%s",
                    "name":"%s",
                    "surname":"%s",
                    "birthdate":"%s",
                    "country_code":"%s"
                }"""
                .formatted(uuid, name, surname, birthdate, countryCode);
    }

    public String toFullJson() {
        return """
                {
                    "uuid":"%s",
                    "name":"%s",
                    "surname":"%s",
                    "birthdate":"%s",
                    "country_code":"%s",
                    "bio":"%s",
                    "promises":%s
                }"""
                .formatted(uuid, name, surname, birthdate, countryCode, bio, promises);
    }
}

File: frontend/lib/model/Subject.dart
import 'Promise.dart';

class Subject {
  final String uuid;
  final String name;
  final String surname;
  final String birthdate;
  final String countryCode;
  final String? bio;
  final List<Promise>? promises;

  const Subject({
    required this.uuid,
    required this.name,
    required this.surname,
    required this.birthdate,
    required this.countryCode,
    required this.bio,
    required this.promises,
  });

  factory Subject.fromJson(Map<String, dynamic> json) {
    return Subject(
      uuid: json['uuid'],
      name: json['name'],
      surname: json['surname'],
      birthdate: json['birthdate'],
      countryCode: json['country_code'],
      bio: json['bio'],
      promises: inferPromises(json["promises"]),
    );
  }

  static List<Promise>? inferPromises(List? jsonPromises) {
    return jsonPromises?.map((promise) => Promise.fromJson(promise)).toList();
  }

  @override
  String toString() {
    return '$name $surname, $birthdate, $countryCode';
  }
}
